version: "3.9"

services:

  postgres-db:
    image: postgres:16.2
    restart: always
    env_file:
      - .env.docker
    environment:
      POSTGRES_USER: "postgres"
    container_name: postgres-16-2
    ports:
      - "5433:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U $POSTGRES_USER" ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - personal-diary-network

  web:
    build:
      context: .
      dockerfile: Dockerfile
    command: sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    env_file:
      - .env.docker
    environment:
      ENVIRONMENT: "docker"
    volumes:
      - .:/app/
      - static_volume:/app/static
      - media_volume:/app/media
    ports:
      - "8000:8000"
    depends_on:
      - postgres-db
    networks:
      - personal-diary-network

volumes:
  postgres_data:
  static_volume:
  media_volume:


networks:
  personal-diary-network:
    driver: bridge